hail/python/hailtop/batch/batch.py

```diff
@@ -109,7 +103,6 @@ def __init__(self,
                  default_timeout: Optional[Union[float, int]] = None,
                  default_shell: Optional[str] = None,
                  default_python_image: Optional[str] = None,
-                 project: Optional[str] = None,
```

Comment:
Hmm. True, but there's no safe way to use this argument, there are always cases where an exception will happen before I can close it because Batch doesn't have a close method. I've restored it with a warning and a best effort to close it.

Reply:
I'd almost consider this a bug fix not breaking backwards compatibility 🤷